app/script/view_model/ConversationInputEmojiViewModel.js

```diff
@@ -117,25 +172,31 @@ z.ViewModel.ConversationInputEmojiViewModel = class ConversationInputEmojiViewMo
     if (query.length < QUERY_MIN_LENGTH || query[0] === ' ' || !this.emoji_dict) {
       this.emoji_list.remove();
     } else {
-      const emoji_matched = this.emoji_dict
-        .filter((emoji) => emoji.toLowerCase().indexOf(query) !== -1)
+      const query_words = query.split(' ');
+      const emoji_matched = Object.values(this.emoji_dict)
+        .filter(emoji => {
+          const emoji_name_words = emoji.name.split(' ');
+          return query_words.every(query_word => emoji_name_words.some(emoji_name_word => emoji_name_word.startsWith(query_word)));
+        })
         .sort((emoji_a, emoji_b) => {
-          const [, emoji_name_a] = emoji_a.split('\t');
-          const [, emoji_name_b] = emoji_b.split('\t');
-          return z.util.StringUtil.sort_by_priority(emoji_name_a, emoji_name_b, query);
+          const usage_count_a = this.get_usage_count(emoji_a.name);
+          const usage_count_b = this.get_usage_count(emoji_b.name);
+          if (usage_count_a > usage_count_b) {
+            return -1;
+          } else if (usage_count_a < usage_count_b) {
+            return 1;
+          } else {
+            return z.util.StringUtil.sort_by_priority(emoji_a.name, emoji_b.name, query);
+          }
```

Comment:
Good point, this code is introduced in #1126 so I fixed it there :+1: 
The diff in this PR will be much smaller once all the earlier PRs are merged 🙂